processor/src/chiplets/hasher/trace.rs

```diff
@@ -122,7 +122,6 @@ impl HasherTrace {
             self.append_row(selectors, &hasher_state, node_index);
         }
 
-        // TODO: remove this after the state is removed from the HasherLookup struct
```

Comment:
Do we still need the `state` parameter and the code below? The point of this comment was that I thought we wouldn't need these once the state was removed from the lookups, since this method appends the rows to the trace directly. Have I missed something?

Reply:
I think we need it to get the final state to check that the resultant state digest is the same as the expected hash.
I thought your comment was to avoid copying the state for intermediate states after removing lookups but since we only copy the trace once at the end in each case now, we only copy the final state.

Reply:
Hmm yeah, I see. However, the `result` (which comes from the digest in this state array) and `expected_hash` are only checked in debug mode, which means this state is only used in debug mode, so I'm not sure we want to be passing it around and going through this extra work when not in debug.

I think what we want is to move that `debug_assert_eq` from the chiplets module into the hasher module. We would then check the assertion in debug mode in 2 cases:
1. after building the trace for the first time (in the control block merge methods before saving the memoized version)
2. after copying the trace (inside the `copy_trace` method)

@bobbinth what are your thoughts?

If we do this, we can do it as a separate PR so this one can be merged